subsys/fs/nvs/nvs.c

```diff
@@ -277,6 +277,48 @@ static int nvs_ate_cmp_const(const struct nvs_ate *entry, uint8_t value)
 	return 0;
 }
 
+/* nvs_ate_valid validates an ate:
+ *     return 1 if crc8 and offset valid,
+ *            0 otherwise
+ */
+static int nvs_ate_valid(struct nvs_fs *fs, const struct nvs_ate *entry)
+{
+	size_t ate_size;
+
+	ate_size = nvs_al_size(fs, sizeof(struct nvs_ate));
+
+	if ((nvs_ate_crc8_check(entry)) ||
+	    (entry->offset >= (fs->sector_size - ate_size))) {
+		return 0;
+	}
+
+	return 1;
+
+}
+
+/* nvs_close_ate_valid validates an sector close ate: a valid sector close ate:
+ * - valid ate
+ * - len = 0 and id = 0xFFFF
+ * - offset points to location at ate multiple from sector size
+ * return 1 if valid, 0 otherwise
+ */
+static int nvs_close_ate_valid(struct nvs_fs *fs, const struct nvs_ate *entry)
+{
+	size_t ate_size;
+
+	if ((!nvs_ate_valid(fs, entry)) || (entry->len != 0U) ||
+	    (entry->id != 0xFFFF)) {
```

Comment:
This probably will fit in one 100 character line.

Reply:
Yes, but don't turn the rule around because then I would have to do lot's of changes :-) :-). I will leave this one as is and still try to keep the 80 char rule.

BTW, I think there is a 78 char rule on the PR description in github which already makes it very difficult to copy part of your code into the PR (e.g. as source for discussion).